src/pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -237,40 +246,80 @@ def _adjust_batch_size(
 
     Args:
         trainer: instance of pytorch_lightning.Trainer
-
-        batch_arg_name: name of the field where batch_size is stored.
-
         factor: value which the old batch size is multiplied by to get the
             new batch size
-
         value: if a value is given, will override the batch size with this value.
             Note that the value of `factor` will not have an effect in this case
-
-        desc: either `succeeded` or `failed`. Used purely for logging
+        desc: either ``"succeeded"`` or ``"failed"``. Used purely for logging
 
     Returns:
         The new batch size for the next trial and a bool that signals whether the
         new value is different than the previous batch size.
     """
     model = trainer.lightning_module
     batch_size = lightning_getattr(model, batch_arg_name)
+    assert batch_size is not None
     new_size = value if value is not None else int(batch_size * factor)
     if desc:
-        log.info(f"Batch size {batch_size} {desc}, trying batch size {new_size}")
+        rank_zero_info(f"Batch size {batch_size} {desc}, trying batch size {new_size}")
 
-    if not _is_valid_batch_size(new_size, trainer.train_dataloader, trainer):
-        new_size = min(new_size, len(trainer.train_dataloader.dataset))
+    # TODO improve this for multi eval dataloaders
+    if trainer.state.fn == "fit":
+        if trainer.train_dataloader is None:
+            trainer.reset_train_dataloader()
+
+        assert trainer.train_dataloader is not None
+        if not _is_valid_batch_size(new_size, trainer.train_dataloader, trainer):
+            new_size = min(new_size, len(trainer.train_dataloader.dataset))
+    else:
+        stage = trainer.state.stage
+        assert stage is not None
+        dataloaders = getattr(trainer, f"{stage.dataloader_prefix}_dataloaders")
+        if dataloaders is None:
+            getattr(trainer, f"reset_{stage.dataloader_prefix}_dataloader")()
```

Comment:
Shouldn't you use the `_reset_dataloaders` utility you added below?

Reply:
there's a TODO. Right now it's checked only for train, so no need to use that one.

Reply:
Which TODO? The one in L269?

What is only checked for train?

Couldn't this do:


```python
_reset_dataloaders(trainer, model)
if trainer.state.fn == "fit":
    assert trainer.train_dataloader is not None
    if not _is_valid_batch_size(new_size, trainer.train_dataloader, trainer):
        new_size = min(new_size, len(trainer.train_dataloader.dataset))
else:
    assert dataloaders is not None
    if not _is_valid_batch_size(new_size, dataloaders[0], trainer):
        new_size = min(new_size, len(dataloaders[0].dataset))
```

Reply:
during fit, we can have both val and train dataloaders, but we can't tell whether users are tuning for val_Dataloaders too, or not. So, for now, I have not updated the logic as before, and we only track the total batch size for train dataloaders during fit, but this is a design ques, should we consider val dataloaders here or not?

also, it should not reload the dataloaders if they are already reloaded. For eg in the case of the fine-tuning stage.

Reply:
updated the TODOs